test/cpp/end2end/test_service_impl.cc

```diff
@@ -34,8 +34,22 @@ using std::chrono::system_clock;
 
 namespace grpc {
 namespace testing {
-namespace {
 
+int MetadataMatchCount(
```

Comment:
If this free function is now being used by both client and server code, it should probably be refactored into a separate file called `utils_common.cc` or something like that in this directory.